src/k8s/pkg/k8sd/features/cilium/loadbalancer_test.go

```diff
@@ -0,0 +1,239 @@
+				},
+			},
+			&v1.DaemonSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "cilium",
+					Namespace: "kube-system",
+				},
+			},
+		)
+		fd, ok := clientset.Discovery().(*fakediscovery.FakeDiscovery)
+		g.Expect(ok).To(BeTrue())
+		fd.Resources = []*metav1.APIResourceList{
+			{
+				GroupVersion: "cilium.io/v2alpha1",
+				APIResources: []metav1.APIResource{
+					{Name: "ciliuml2announcementpolicies"},
+					{Name: "ciliumloadbalancerippools"},
+					{Name: "ciliumbgppeeringpolicies"},
+				},
+			},
+		}
+		snapM := &snapmock.Snap{
+			Mock: snapmock.Mock{
+				HelmClient:       helmM,
+				KubernetesClient: &kubernetes.Client{Interface: clientset},
+			},
+		}
+		lbCfg := types.LoadBalancer{
+			Enabled: ptr.To(true),
+			// setting both modes to true for testing purposes
+			L2Mode:         ptr.To(true),
+			L2Interfaces:   ptr.To([]string{"eth0", "eth1"}),
+			BGPMode:        ptr.To(true),
+			BGPLocalASN:    ptr.To(64512),
+			BGPPeerAddress: ptr.To("10.0.0.1/32"),
+			BGPPeerASN:     ptr.To(64513),
+			BGPPeerPort:    ptr.To(179),
+			CIDRs:          ptr.To([]string{"192.0.2.0/24"}),
+			IPRanges: ptr.To([]types.LoadBalancer_IPRange{
+				{Start: "20.0.20.100", Stop: "20.0.20.200"},
+			}),
+		}
+		networkCfg := types.Network{
+			Enabled: ptr.To(true),
+		}
+
+		status, err := cilium.ApplyLoadBalancer(context.Background(), snapM, lbCfg, networkCfg, nil)
+
+		g.Expect(err).ToNot(HaveOccurred())
+		g.Expect(status.Enabled).To(BeTrue())
+		g.Expect(status.Version).To(Equal(cilium.CiliumAgentImageTag))
+
+		g.Expect(helmM.ApplyCalledWith).To(HaveLen(2))
+
+		firstCallArgs := helmM.ApplyCalledWith[0]
+		g.Expect(firstCallArgs.Chart).To(Equal(cilium.ChartCilium))
+		g.Expect(firstCallArgs.State).To(Equal(helm.StateUpgradeOnlyOrDeleted(networkCfg.GetEnabled())))
+		g.Expect(firstCallArgs.Values["l2announcements"].(map[string]any)["enabled"]).To(Equal(lbCfg.GetL2Mode()))
+		g.Expect(firstCallArgs.Values["bgpControlPlane"].(map[string]any)["enabled"]).To(Equal(lbCfg.GetL2Mode()))
+
+		secondCallArgs := helmM.ApplyCalledWith[1]
+		g.Expect(secondCallArgs.Chart).To(Equal(cilium.ChartCiliumLoadBalancer))
+		g.Expect(secondCallArgs.State).To(Equal(helm.StatePresent))
+		validateLBValues(t, secondCallArgs.Values, lbCfg)
+
+		// check if cilium-operator and cilium daemonset are restarted
+		deployment, err := clientset.AppsV1().Deployments("kube-system").Get(context.Background(), "cilium-operator", metav1.GetOptions{})
+		g.Expect(err).ToNot(HaveOccurred())
+		g.Expect(deployment.Spec.Template.Annotations).To(HaveKey("kubectl.kubernetes.io/restartedAt"))
+		daemonSet, err := clientset.AppsV1().DaemonSets("kube-system").Get(context.Background(), "cilium", metav1.GetOptions{})
+		g.Expect(err).ToNot(HaveOccurred())
+		g.Expect(daemonSet.Spec.Template.Annotations).To(HaveKey("kubectl.kubernetes.io/restartedAt"))
+	})
+}
+
+func validateLBValues(t *testing.T, values map[string]interface{}, lbCfg types.LoadBalancer) {
```

Comment:
nit: in the metallb test you call this `validateLoadBalancerValues`. Let's be consistent on either option